### DISCUSSION TITLE
Displaying png files in part screen.

### DISCUSSION THREAD

**Discussion by ColinHMurdoch** [Bodmer/TFT_eSPI#1723] [General]
Hi Bodmer,

I am just upgrading a project I wrote some time ago.  I wish to draw some images in area of the screen, and I am wondering what is the best technique before I start.  I am using a 480x320 TFT (ILI9488) and wish to fill about 1/4 of the screen with the image and normal text around it.  Should I reduce the images to the size of this area in pixels, or should I try and use the viewport facility and draw the images inside it.  Your advice would be appreciated.
Regards,
Colin Murdoch.

**Comment by ColinHMurdoch** [comment]
Hi Bodmer,
I think perhaps I have answered my own question.  I have just found your ESP32_SDcard_jpeg.ino sketch which draws jpg files stored on the SD card. As my display has an SC card built in (which I have not tested yet) I think perhaps this would be the easiest way for me to go.  I should be able to use this as a basis for drawing the images in the top left of the screen (position 0 , 0 I think?). As long as I size all the images the same then I should be able to draw them on 1/4 of the screen.
I'll have a play with this and give it a try.
Regards,
Colin Murdoch.

**Reply by Bodmer** [reply]
PNG compressed images are also an option if you can spare ~40kbytes of RAM, examples have been added.

Images can be positioned as you wish, just change the [x,y corrdinate](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Generic/ESP32_SDcard_jpeg/ESP32_SDcard_jpeg.ino#L80) for the top left corner.

**Comment by paulw517** [comment]
I  am attempting to compile Bodmer's sketch LittleFS_PNG.ino, however, it returns the error messages below. 

https://github.com/Bodmer/TFT_eSPI/blob/master/examples/PNG%20Images/LittleFS_PNG/LittleFS_PNG.ino

I am working with Arduino IDE 1.8.12 and an Elecrow 3.5 in ILI9488 display with an on board ESP32-Wrover-B micro.  

``
C:\Users\Paul\Documents\Arduino\Sketches\LittleFS_PNG_2\LittleFS_PNG_2.ino: In function 'void loop()':
LittleFS_PNG_2:69:46: error: 'pngOpen' was not declared in this scope
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                              ^~~~~~~
C:\Users\Paul\Documents\Arduino\Sketches\LittleFS_PNG_2\LittleFS_PNG_2.ino:69:46: note: suggested alternative: 'popen'
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                              ^~~~~~~
                                              popen
LittleFS_PNG_2:69:55: error: 'pngClose' was not declared in this scope
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                                       ^~~~~~~~
C:\Users\Paul\Documents\Arduino\Sketches\LittleFS_PNG_2\LittleFS_PNG_2.ino:69:55: note: suggested alternative: 'pclose'
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                                       ^~~~~~~~
                                                       pclose
LittleFS_PNG_2:69:65: error: 'pngRead' was not declared in this scope
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                                                 ^~~~~~~
C:\Users\Paul\Documents\Arduino\Sketches\LittleFS_PNG_2\LittleFS_PNG_2.ino:69:65: note: suggested alternative: 'pngDraw'
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                                                 ^~~~~~~
                                                                 pngDraw
LittleFS_PNG_2:69:74: error: 'pngSeek' was not declared in this scope
       int16_t rc = png.open(strname.c_str(), pngOpen, pngClose, pngRead, pngSeek, pngDraw);
                                                                          ^~~~~~~
``
I have achieved displaying jpeg images on my Elecrow display, I just need to overcome the above issuses with png images.
